src/main/java/duke/command/MyParser.java

```diff
@@ -0,0 +1,134 @@
+                throw new DukeException("OOPS!!! The description of delete must be an integer");
+            }
+
+            int targetTask = Integer.parseInt(userDescription);
+            duke.dukeDelete(targetTask);
+            break;
+        }
+        case "todo":
+            if (userDescription.isBlank()) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty");
+            }
+
+            duke.dukeTodo(userDescription);
+            break;
+        case "deadline": {
+            int dueByIndex = userDescription.indexOf("/by ");
+            if (userDescription.isBlank() || dueByIndex == 0) {
+                throw new DukeException("OOPS!!! The description of a deadline cannot be empty");
+            }
+
+            if (dueByIndex == -1) {
+                throw new DukeException("OOPS!!! No deadline has been set. Input with '/by dd/mm/yyyy 0000'");
+            }
+
+            String desc = userDescription.substring(0, dueByIndex);
+            String dueBy = userDescription.substring(dueByIndex + 4);
+
+            Calendar deadlineCal = Calendar.getInstance();
+            SimpleDateFormat deadlineTimeFormat = new SimpleDateFormat("d/MM/yyyy HHmm");
+            Date d;
+            try {
+                d = deadlineTimeFormat.parse(dueBy);
+                deadlineCal.setTime(d);
+            } catch (ParseException e) {
+                throw new DukeException("OOPS!!! The date is not formatted as dd/mm/yyyy 0000");
+            }
+            duke.dukeDeadline(desc, deadlineCal);
+            break;
+        }
+        case "event": {
+            int atIndex = userDescription.indexOf("/at ");
+            if (userDescription.isBlank() || atIndex == 0) {
+                throw new DukeException("OOPS!!! The description of an event cannot be empty");
+            }
+
+            if (atIndex == -1) {
+                throw new DukeException("OOPS!!! No event time has been set. Input with '/at dd/mm/yyyy 0000'");
+            }
+
+            String desc = userDescription.substring(0, atIndex);
+            String at = userDescription.substring(atIndex + 4);
+
+            Calendar eventCal = Calendar.getInstance();
+            SimpleDateFormat eventTimeFormat = new SimpleDateFormat("d/MM/yyyy HHmm");
+
+            try {
+                eventCal.setTime(eventTimeFormat.parse(at));
+            } catch (ParseException e) {
+                throw new DukeException("OOPS!!! The date is not formatted as dd/mm/yyyy 0000");
+            }
+            duke.dukeEvent(desc, eventCal);
+            break;
+        }
+        case "find":
+            if (userDescription.isBlank()) {
+                throw new DukeException("OOPS!!! The description of find cannot be empty");
+            }
+
+            duke.dukeFind(userDescription);
+            break;
+        default:
+            throw new DukeException("OOPS!!! I'm Sorry, but I don't know what that means");
+        }
+    }
+}
+
```

Comment:
Maybe this method is a bit too long? Maybe you can break down each case to separate methods?